core/CMakeLists.txt

```diff
@@ -93,8 +93,10 @@ cubos_common_target_options(cubos-core)
 # Link dependencies
 
 if (WITH_OPENGL)
-    add_subdirectory(lib/glad)
-    target_link_libraries(cubos-core PRIVATE glad)
+    set(GLAD_SOUURCES_DIR "lib/glad")
+    add_subdirectory("${GLAD_SOUURCES_DIR}/cmake" glad_cmake SYSTEM)
+    glad_add_library(glad REPRODUCIBLE API gl:core=3.3)
+    target_link_libraries(cubos-core PUBLIC glad)
```

Comment:
Why PUBLIC instead of PRIVATE?

Reply:
Just copied it from the glad documentation, I don't actually know how cmake works

Reply:
By making it PUBLIC you're making dependencies of CUBOS. have it on their include dir too.
Previously it was PRIVATE, try setting it back and see if anything breaks.